src/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -860,14 +860,34 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_G()
             Assert.Equal(expectedString, result.ToString("g"));
         }
 
-        [Fact]
-        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
+        [Theory, MemberData("Format_String_TestData_O")]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O(DateTimeOffset dt, string expected)
```

Comment:
You might start the test by adding:
```C#
Assert.Equal(expected, dt.ToString("o");
```
which validates that the provided dt does in fact ToString to the supplied expected string.  Then you can continue with what you have, which further validates that that rendered string can be parsed and that result ToString'd again back to the expected string, i.e. that it roundtrips, e.g.
```C#
string actual = dt.ToString("o");
Assert.Equal(expected, actual);

DateTimeOffset parsed = DateTimeOffset.ParseExact(actual, "o", null, DateTimeStyles.None);
Assert.Equal(dt, parsed);
Assert.Equal(expected, parsed.ToString("o");
```
(assuming all of those statements are true :)